src/vocabularyChecker/detectWrongSyntax.ts

```diff
@@ -0,0 +1,100 @@
+import json from "./AllowedVocab.json";
+/**
+ * Interface containing @param fromIndex: number, @param toIndex: number, @param message: string
+ */
+import Issue from "./Issue";
+/**
+ * Regular expression that looks for any and all words. A word is defined as some sequence of
+ * characters (\w*S) betweeen wordbarriers (\b)
+ */
+const anyWord = new RegExp(/\b(\w*\S)\b/g);
+/**
+ * Predefined dictionary containing all allowed Words
+ */
+const allowedWords = json;
+interface Position { fromIndex: number; toIndex: number; }
+
+/**
+ * @param text
+ * @param invalidWords
+ * Returns a map of invalid words (consisted in @param invalidWords) and their corresponding
+ * positions (starting position and end position) inside of a string.
+ */
+
+export function createMapOfInvalidWords(text: string, invalidWords: string[]): Map<string, [number, number]> {
+	const positions = new Map<string, [number, number]>();
+	for (const word of invalidWords) {
+		positions.set(word, [text.indexOf(word), text.indexOf(word) + word.length - 1]);
+	}
+	return positions;
+}
+
+/**
+ * @param text
+ * @param invalidWords
+ * Returns message that tells the user which words are not allowed and on which
+ * position those are within @param text.
+ */
+export function createErrorMessages(text: string, invalidWords: string[]): string|undefined {
+	const wrongWords = logInvalidWords(createMapOfInvalidWords(text, invalidWords));
+	return wrongWords;
+}
+
+/**
+ * @param text
+ * Just a function to bundle the functionalities of @function collectAllInvalidWords and
```

Comment:
Comments generally shouldn't describe how a function works internally ("first calls X, then calls Y, and returns Z"), but rather describe what the function should be used for / any edge cases to look out for. Usually it's best to choose function names that already sufficiently describe what the function does, so that comments are not necessary, as they are likely to get out of sync when the code they describe gets changed.